src/Symfony/Component/Mailer/Transport/Smtp/Auth/AuthenticatorInterface.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Mailer\Transport\Smtp\Auth;
+
+use Symfony\Component\Mailer\Transport\Smtp\EsmtpTransport;
+
+/**
+ * An Authentication mechanism.
+ *
+ * @author Chris Corbyn
+ *
+ * @experimental in 4.3
+ */
+interface AuthenticatorInterface
+{
+    /**
+     * Tries to authenticate the user with $username and $password.
```

Comment:
Where does `$username` and `$password` come from?
And what if the authentication failed, does it want an exception being thrown?

Reply:
Removed the obsolete phpdoc message. Added a `@throws` annotation.